test/schema/home/home_page_artist_modules.js

```diff
@@ -24,14 +23,18 @@ describe('HomePageArtistModules', () => {
       beforeEach(() => {
         suggestions = [];
 
-        const gravity = sinon.stub();
-        gravity.with = sinon.stub().returns(gravity);
-        gravity.returns(Promise.resolve(suggestions));
-        HomePageArtistModule.__Rewire__('gravity', gravity);
+        HomePageArtistModule.__Rewire__('gravity', () => {
+          return { with: () => Promise.resolve(suggestions) };
+        });
+
+        HomePageArtistModule.__Rewire__('total', () => {
+          return Promise.resolve({ body: { total: suggestions.length } });
+        });
```

Comment:
I wasn’t about to find a way with sinon to return a dynamic value (the length of the `suggestions` array _at the time of calling the function_), after wasting a bunch of time I figured that using plain objects and functions was much clearer anyways. So no need for sinon in such cases.